modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -542,60 +574,52 @@ public FlowConstruct getFlowConstruct() {
-        initialiseIfNeeded(sourceAdapter);
-        if (restarting) {
-          sourceAdapter.finishRestart(restartContext);
-        }
-        sourceAdapter.start();
-      } catch (Exception e) {
-        try {
-          // On connection exception, if the failed connection is present, it must be invalidated before stopping the source. This
-          // warranties that a possible call to connectionProvider.disconnect made on the onStop method of the source, does not
-          // affect the connection's invalidation
-          extractConnectionException(e).ifPresent(connectionException -> invalidateConnection(connectionException));
-          stopSource();
-        } catch (Exception eStop) {
-          e.addSuppressed(eStop);
-        }
-        try {
-          disposeSource();
-        } catch (Exception eDispose) {
-          e.addSuppressed(eDispose);
-        }
-        Throwable throwable = exceptionEnricherManager.process(e);
-        Optional<ConnectionException> connectionException = extractConnectionException(throwable);
-        if (connectionException.isPresent()) {
-          throwable = connectionException.get();
-        }
-        throw throwable instanceof Exception ? ((Exception) throwable) : new MuleRuntimeException(throwable);
+        // On connection exception, if the failed connection is present, it must be invalidated before stopping the source. This
+        // warranties that a possible call to connectionProvider.disconnect made on the onStop method of the source, does not
+        // affect the connection's invalidation
+        extractConnectionException(e).ifPresent(connectionException -> invalidateConnection(connectionException));
+        stopSource();
+      } catch (Exception eStop) {
+        e.addSuppressed(eStop);
+      }
+      try {
+        disposeSource();
+      } catch (Exception eDispose) {
+        e.addSuppressed(eDispose);
+      }
+      Throwable throwable = exceptionEnricherManager.process(e);
+      Optional<ConnectionException> connectionException = extractConnectionException(throwable);
+      if (connectionException.isPresent()) {
+        throwable = connectionException.get();
       }
+      throwable = throwable instanceof Exception ? ((Exception) throwable) : new MuleRuntimeException(throwable);
+      future.completeExceptionally(throwable);
     }
+  }
 
-    @Override
-    public String getWorkDescription() {
-      return "Message Source Reconnection";
-    }
+  private void computeStats() {
+    AllStatistics statistics = muleContext.getStatistics();
 
-    @Override
-    public Object getWorkOwner() {
-      return ExtensionMessageSource.this;
+    if (statistics != null && statistics.isEnabled() && computeConnectionErrorsInStats()) {
+      statistics.getApplicationStatistics().incConnectionErrors();
     }
   }
 
+  private boolean computeConnectionErrorsInStats() {
+    return featureFlaggingService.isEnabled(COMPUTE_CONNECTION_ERRORS_IN_STATS);
```

Comment:
why? What I would expect from a feature flag is recovering the current behavior rather than enabling this

Reply:
what do you mean? This is just checking if the feature isEnabled or not, it is not enabling it